src/NSubstitute/Core/SubstitutionContext.cs

```diff
@@ -151,13 +151,23 @@ public PendingSpecificationInfo PendingSpecificationInfo
             ThreadContext.UsePendingRaisingEventArgumentsFactory();
 
         [Obsolete("This method is obsolete and will be removed in a future version of the product. " +
-                  "Use the " + nameof(ThreadContext) + "." + nameof(IThreadLocalContext.AddToQuery) + "() method instead.")]
-        public void AddToQuery(object target, ICallSpecification callSpecification) =>
-            ThreadContext.AddToQuery(target, callSpecification);
+                  "Use the " + nameof(ThreadContext) + "." + nameof(IThreadLocalContext.RegisterInContextQuery) + "() method instead.")]
+        public void AddToQuery(object target, ICallSpecification callSpecification)
+        {
+            // We cannot simulate the API anymore as it changed drastically.
+            // That should be fine, as this method was expected to be called from the NSubstitute core only.
+            throw new NotSupportedException(
```

Comment:
I'd prefer this as a compile-time failure and added to BreakingChanges list. Or do you need API compatibility to work across multiple versions of the library?

Reply:
@dtchepak Updated the code to make it a compile time error (made `Obsolete(error: true)`). Thanks for the hint!

The reason why I don't want to remove API is that it could happen that somebody used this method, so obsolete message will be a hint for them what to do. You might say that I perform a lot of other internal refactorings and don't worry about users there and you will be right. However, the reason why I'm so meticulous here is that `SubstitutionContext` is a facade, an entry point for a lot of APIs, therefore there are large chances that it's used by the clients. I want their migration to be as smooth as possible.